app/src/main/java/uk/gov/onelogin/mainnav/ui/MainNavScreen.kt

```diff
@@ -0,0 +1,105 @@
+package uk.gov.onelogin.mainnav.ui
+
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.NavigationBarItemDefaults
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.TextUnit
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavGraph.Companion.findStartDestination
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import uk.gov.onelogin.R
+import uk.gov.onelogin.mainnav.nav.BottomNavDestination
+import uk.gov.onelogin.ui.components.SimpleTextPage
+import uk.gov.onelogin.ui.home.HomeScreen
+import uk.gov.ui.components.navigation.GdsNavigationBar
+import uk.gov.ui.components.navigation.GdsNavigationItem
+
+@Suppress("LongMethod")
+@Composable
+fun MainNavScreen(
+    navController: NavHostController = rememberNavController()
```

Comment:
Do you want a new navController?
Can this new navController reach some destinations defined in the main one. Like generic error screens? etc

Reply:
see above comment about this